test/e2e/specs/all-features/record/related-table.spec.ts

```diff
@@ -831,6 +836,36 @@ test.describe('Related tables', () => {
     });
   });
 
+  /**
+   * The following 2 tests are for testing the prefill functionality when the inbound foreign key is part of a table that is "almost" pure and binary
+   *
+   * This means there are 2 foreign keys that are part of the same key (making the pair unique) and there are other columns that are not foreign keys
+   */
+  test.describe('for a table that is almost pure and binary and the foreign keys are a unique key', async () => {
```

Comment:
From the comment above, there was a bug regarding selecting leaf values and removing forms that was fixed.

I need to update the test cases to ALSO check for the values that are disabled in the modals/dropdowns and not just the “# of disabled rows”. 

I will also add a test case for the example where the initial modal is “closed” and make sure the first form remains “unfilled” for the “leaf value” since no selections were made.